driver/terraform.go

```diff
@@ -610,6 +609,55 @@ func getTerraformHandlers(taskName string, providers TerraformProviderBlocks) (h
 	return next, nil
 }
 
+// getServicesMetaData helps retrieve metadata which can come from a number of
+// configuration sources: service block, condition block, source_input block.
+// Currently, it is only possible for a task to have metadata defined in one of
+// the listed sources
```

Comment:
Do we error if more than one is defined or does it just have a precedence order?

Reply:
Ah good question! So with some later config streamlining changes, I'm planning to error at the configuration-level so that a task cannot be configured with more than one of `task.services`, `condition "services"`, `source_input "services"`. Ideally if more than one metadata is defined, it should error before reaching this method. I should have shared this context and can update the comment to clarify this!

To actually answer your question, if somehow more than one metadata is defined, this implementation is just precedence of order. My initial thought is that it would be ok to not error here because of the configuration validation, but I might be overlooking something. Please let me know if you have thoughts or concerns!

Reply:
That all makes sense to me, thanks for clarifying!

Reply:
Thanks for checking me! I will add a comment with this clarifying details